### PR TITLE
Drop any kymatio imports

### PR DESCRIPTION
**Pull request by Chris-Pedersen**
Given we want to build statistical isotropy into the model and have more freedom in the numbers of first and second order filters, we will want more flexibility in generating the wavelets and performing the convolutions than Kymatio is intended for. So in this PR we want to take the key computational parts of Kymatio (i.e. the scattering convolutions) and integrate them into our code, and detatch from any other Kymatio imports. This first commit is a crude first stab at this, but there's a lot more tidying to be done. A few major points:

1. There's duplicate code between Kymatio and the parametric scattering network (PSN) code to generate wavelet filters. `filter_bank(..)` from Kymatio is called, followed by `raw_morlet(..)` from PSN. We should only need one of these, but in rewriting this, make sure that the parameters are properly included in the backprop.
2. Also duplicate code for things like periodising filters which appear in both `kymat_code` and `scattering2d` that can be removed.
3. Consider if we want to keep padding. We don't want reflective padding for CAMELs applications, but it might be worth keeping the code in so the model can be applied to other datasets without too much extra work.

### CODE REVIEW

**Comment by Chris-Pedersen** [comment]
Have also noticed that the `scatteringBase.wavelets` have `requires_grad=True`. I don't think this should be the case - only the `params_filters` are needed for autograd, the wavelet "pixel" values themselves shouldn't be included.

**Comment by Chris-Pedersen** [comment]
Ok getting there. Miles suggested not bothering with buffering the wavelet tensors, and just extending the `model.to(device)` inherited method to move them when the device changes. This seems like the best solution, so I've implemented that, and cut out almost all of the remaining kymatio code.

Our smoothing filters are currently generated using the kymatio `gabor_2d` code, and this is now it's only purpose in the code. So next steps are:

1. Remove this and just have a Gaussian filter generated from an appropriate function
2. Tidy up the comments and docstrings

Then I think we are good to merge this branch.

**Comment by Chris-Pedersen** [comment]
Have tested this reproduces old results. Still a bit more tidying up to go but probably a good place for a merge.
